daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/InitiatedContentParsers.scala

```diff
@@ -47,13 +57,18 @@ final class InitiatedContentDiscrimChoiceAndIndexGreaterThanMinParser(
   extends PrimParser {
   override lazy val runtimeDependencies = Vector()
 
-  def parse(start: PState): Unit = {
-    val top = start.discriminator
-    start.popPointOfUncertainty // pop array discriminator off of stack and any changed variables
-    start.setDiscriminator(true) // set the choice discriminator
-    start.pushPointOfUncertainty
-    start.setDiscriminator(top) // restore the array discriminator to top of stack
-    if (start.arrayPos > min)
-      start.setDiscriminator(true) // set array discriminator only if
+  final def parse(start: PState): Unit = {
+    if (start.arrayPos > min) {
+      // resolve the point of uncertainty associated with array elements once
+      // we have parsed min occurrences. There should not be a point of
+      // uncertainty associated with this array until we have parsed more than
+      // min occurrences
+      start.resolvePointOfUncertainty()
+    }
+
+    if (start.arrayPos == 1) {
```

Comment:
So if min is 0 and start.arrayPos is 1, I think this logic calls start.resolvePointOfUncertainty() twice, which can't be right can it?

Reply:
If min is zero, i think there should be a point of uncertainty for the first first array element, since the entire array is optional. And there should also be a separate point of uncertainty for the enclosing choice. So this should resolve two points of uncertainty. At least that's how I understand this stuff. I'll add a test to make sure this does the right thing.

Reply:
Add a comment here also to save people taking it out, only to find it breaks the test, and putting it back in. Worth it to say there really are two PoUs here.